ops/testing.py

```diff
@@ -250,6 +253,29 @@ def _create_meta(self, charm_metadata, action_metadata):
 
         return charm.CharmMeta.from_yaml(charm_metadata, action_metadata)
 
+    def _load_config_defaults(self, charm_config):
+        """Load default values from config.yaml
+
+        Handle the case where a user doesn't supply explicit config snippets.
+        """
+        filename = inspect.getfile(self._charm_cls)
+        charm_dir = pathlib.Path(filename).parents[1]
+
+        if charm_config is None:
+            config_path = charm_dir / 'config.yaml'
+            if config_path.is_file():
+                charm_config = config_path.read_text()
+                self._charm_dir = charm_dir
+            else:
+                # The simplest of config that the framework can support
+                charm_config = '{}'
+        elif isinstance(charm_config, str):
+            charm_config = dedent(charm_config)
+        charm_config = yaml.load(charm_config, Loader=yaml.SafeLoader)
+        charm_config = charm_config.get('options', {})
+        return {key: value['default'] for key, value in charm_config.items()
+                if 'default' in value}
```

Comment:
The rest looks great, but why are you filtering out the rest of the information? Maybe get whatever is under `options` is fine (if `options` is the only key, I'd need to search for it), but why losing `type`, or `description`?

Thanks!!

Reply:
config in ops has no description, and the yaml will already load with the right type (usually), right?

Reply:
The result of this is being passed to `Harness.update_config` which takes a mapping of `key: value`.  So this converts it to that format.  We also don't want to set any items that have no default value, but I just realized I should add that to the test case.